### DISCUSSION TITLE
'devbuild' process failing

### DISCUSSION THREAD

**Discussion by mosBrkr** [wixtoolset/issues#7368] [WiX Development]
Hey guys

I tried to build the WiX Toolset v4 following the description in the [README](https://github.com/wixtoolset/wix4#readme) but unfortunately I get several warnings and errors during the build process:

> ...\wix4\src\internal\WixInternal.BaseBuildTasks.Sources\WixInternal.BaseBuildTasks.Sources.csproj : error NU1604: Project dependency Microsoft.SourceLink.GitHub does not contain an inclusive lower bound.
 Include a lower bound in the dependency version to ensure consistent restore results.

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency Microsoft.Build.Tasks.Core does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency Microsoft.SourceLink.GitHub does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency WixToolset.Dtf.Compression does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency WixToolset.Dtf.Compression.Cab does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency WixToolset.Dtf.WindowsInstaller does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1604: Project dependency xunit does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [TargetFramework=netstandard2.0]

> ...\wix4\src\internal\WixInternal.TestSupport\WixInternal.TestSupport.csproj : error NU1701: Package 'xunit 1.7.0.1540' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NET Framework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [TargetFramework=netstandard2.0]

Followed by a lot of compiler errors in that project ('WixInternal.TestSupport') about unknown type- or namespacenames.

Do you have an idea what could cause this issue?

One small difference I have seen is that the `MSVC v143 - VS 2022 C++ ARM64 build tools (Latest)` shows up as `MSVC v143 - VS 2022 C++ ARM64/ARM64EC build tools (Latest)` in Visual Studio 2022 - 17.5.3 (already mentioned here: https://github.com/wixtoolset/issues/issues/7331#issue-1648313913)

**Comment by robmen** [comment]
My first guess is that your build tooling is missing support for [Central Package Management](https://learn.microsoft.com/en-us/nuget/consume-packages/Central-Package-Management).

**Comment by mosBrkr** [comment]
Looks like a perfect guess 🙂 

I had to put the directory to the downloaded nuget.exe command-line-tool in front of my dotnet installation on the path.

Now the build finished successfully

Thanks for the quick help!

**Comment by chrpai** [comment]
I just reverifyed the README.md instructions on a clean machine with the latest release of VS and all the component names are still correct. I copy and pasted each into the search to confim.
